kernel/Makefile

```diff
@@ -0,0 +1,169 @@
+### Build configuration ###
+
+# C compiler used to compile kernel C code
+# Compiler built by xbstrap is in its build directory
+CC = ../build/xbstrap-dir/tools/kernel-gcc/bin/x86_64-elf-gcc
+
+# Assembler used to assemble kernel asm files
+AS = nasm
+
+# Linker used to generate kernel binary from object files
+# We use gcc itself for linking
+# TODO: replace with x86_64-lezione-gcc once we have xbstrap config
+LD = $(CC)
+
+# Common C flags (used both in debug and release mode)
+# -c - create object file
+# -mno-80387, -mno-mmx, -mno-sse, -mno-sse2 - do not generate code
+# that uses extended registers (87 fpu, MMX, SSE, SSE2)
+# -mno-red-zone - do not assume red zone 
+# -mcmodel=kernel - instruct gcc that kernel will be located
+# in negative 2 gigabytes of the virtual address space
```

Comment:
maybe specify better that this means that the kernel is loaded at virtual address 0xFFFFFFFFFFFFFFFF - 2M

Reply:
This would be super confusing, as beginners might think that GCC somehow influences the offset at which kernel is loaded